transport/http/transport.go

```diff
@@ -174,16 +218,27 @@ func (a *Transport) RetainPeer(pid peer.Identifier, sub peer.Subscriber) (peer.P
 	return p, nil
 }
 
+// for httpPeer to obtain a peer or nil
+func (a *Transport) getPeerWithLock(addr string) *httpPeer {
+	a.lock.Lock()
+	defer a.lock.Unlock()
+
+	if p, ok := a.peers[addr]; ok {
+		return p
+	}
+	return nil
```

Comment:
optional: You can just `return a.peers[addr]`. It will be nil if it wasn't in the map.